DuckDuckGo/TabURLInterceptor.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  TabURLInterceptor.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+import Common
+import Subscription
```

Comment:
General note on the class:
From my understanding, the rush for the release is over and I would expect every new feature to be Unit Tested.
I realise that this class is very simple, so It should require very simple tests :)